src/SourceBuild/content/repo-projects/installer.proj

```diff
@@ -78,11 +78,20 @@
   <!-- Copy SDK archive to assets root. -->
   <Target Name="CopySdkArchive"
           AfterTargets="Build"
-          DependsOnTargets="DetermineSourceBuiltSdkVersion">
-    <Copy SourceFiles="$(SdkTarballPath)"
+          DependsOnTargets="GetDotNetSdkArchive">
+
+    <Copy SourceFiles="$(SdkArchivePath)"
           DestinationFolder="$(ArtifactsAssetsDir)"
           SkipUnchangedFiles="true"
           UseHardlinksIfPossible="true" />
+
+    <!-- Extract the SDK archive so its contents can be used downstream -->
+    <MakeDir Directories="$(DotNetSdkExtractDir)" />
```

Comment:
Is this always used? (symbol layout and tests). Maybe conditionalize the extraction it if it won't be used later.

Reply:
This is a product dependency for source-build as part of the finish-source-only.proj and ideally also a test dependency by the two test projects. Today those test projects extract the SDK archive yet again and it would be good to consolidate that in a follow-up.

I agree that this should be conditioned to only run when needed. With the new pattern of encapsulating code units we can achieve that and declare the dependency directly.

Meaning, I would put this into a eng/extract-sdk-archive.proj NoTargets msbuild sdk file and then reference that from finish-source-only.proj and the test projects. That's what we do with publish.proj which creates the merged asset manifest.

Reply:
Fixed in https://github.com/dotnet/installer/pull/19222/commits/41b46b4cc39f962914c83d2257e9a2f4b75364e3